src/app/todo/page.tsx

```diff
@@ -0,0 +1,64 @@
+"use client";
+
+import { useRedirect } from "@/hooks/useRedirect";
+import { useUser } from "@/hooks/useUser";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+
+export default function Todo() {
+  const [newTodo, setNewTodo] = useState("");
+  const [todos, setTodos] = useState<any[]>([]);
+  const [metaData, setMetadata] = useState({ owner: "J.Doe", count: 0 });
+
+  const user = useUser();
+
+  if (!user.loggedIn) {
```

Comment:
The useRedirect hook should be used to redirect the user to the login page if they are not logged in.